examples/go-kafkacat/go-kafkacat.go

```diff
@@ -177,6 +265,18 @@ func runConsumer(config *kafka.ConfigMap, topics []string) {
 				if verbosity >= 2 {
 					fmt.Fprintf(os.Stderr, "%% %v\n", e)
 				}
+			case kafka.OAuthBearerTokenRefresh:
+				oauthBearerToken, retrieveErr := retrieveUnsecuredToken(e, config)
```

Comment:
The token refresh handling case for the consumer is basically the same as the producer. Can we use the `Handle` interface to make this a generic helper that both can use?

Reply:
@jamesd I noticed this DRY issue, too.  I looked at it and noticed that the `fmt.Fprintf(os.Stderr,...)` statement is specific to `go-kafkacat`, and the retrieval is definitely something that will differ from one installation to another.  There is also desire to be able to differentiate between a retrieval error (which might be caused by the token endpoint being down) and an error setting the token (which is likely caused by a bad configuration -- for example an illegal extension key or value).  So the only code I can think of that might work is something like this:

```
oauthBearerToken, retrieveErr := retrieveUnsecuredToken(e, config)
if retrieveErr != nil {
	fmt.Fprintf(os.Stderr, "%% Token retrieval error: %v\n", retrieveErr)
}
setTokenError := theHandle.processOAuthBearerTokenRetrieval(oauthBearerToken, retrieveErr)
if setTokenError != nil {
	fmt.Fprintf(os.Stderr, "%% Error setting token and extensions: %v\n", setTokenError)
}
```

I'm not sure if declaring `func (h *Handle) processOAuthBearerTokenRetrieval(oauthBearerToken OAuthBearerToken, retrieveErr error) error` is worth it.  It _could_ eliminate the `SetOAuthBearerToken/SetOAuthBearerTokenFailure()` methods, though I hesitate to remove these building blocks.

Thoughts on this or another, possibly better way to adhere to DRY?

Reply:
That makes sense. At a minimum, how about just a helper in `go-kafkacat`? Something like:
```suggestion
func handleOAuthBearerTokenRefreshEvent(client Handle, e kafka.OAuthBearerTokenRefresh, config *kafka.ConfigMap) {
	oauthBearerToken, retrieveErr := retrieveUnsecuredToken(e, config)
	if retrieveErr != nil {
		fmt.Fprintf(os.Stderr, "%% Token retrieval error: %v\n", retrieveErr)
		client.SetOAuthBearerTokenFailure(retrieveErr.Error())
	} else {
		setTokenError := client.SetOAuthBearerToken(oauthBearerToken)
		if setTokenError != nil {
			fmt.Fprintf(os.Stderr, "%% Error setting token and extensions: %v\n", setTokenError)
			client.SetOAuthBearerTokenFailure(setTokenError.Error())
		}
	}
}
```

Reply:
👍 Done